tests/datasource/test_new_datasource_with_runtime_data_connector.py

```diff
@@ -0,0 +1,388 @@
+        "partition_request": partition_request,
+        "limit": None,
+    }
+    batch_request: BatchRequest = BatchRequest(**batch_request)
+
+    with pytest.raises(ge_exceptions.DataConnectorError):
+        # noinspection PyUnusedLocal
+        batch_list: List[
+            Batch
+        ] = basic_datasource_with_runtime_data_connector.get_batch_list_from_batch_request(
+            batch_request=batch_request
+        )
+    partition_request = {"partition_identifiers": {"unknown_key": "some_value"}}
+
+    # Insure that keys in partition_request["partition_identifiers"] that are not among runtime_keys declared in configuration
+    # are not accepted.  In this test, a single illegal key is present.
+    batch_request: dict = {
+        "datasource_name": basic_datasource_with_runtime_data_connector.name,
+        "data_connector_name": "test_runtime_data_connector",
+        "data_asset_name": "IN_MEMORY_DATA_ASSET",
+        "batch_data": test_df,
+        "partition_request": partition_request,
+        "limit": None,
+    }
+    batch_request: BatchRequest = BatchRequest(**batch_request)
+
+    with pytest.raises(ge_exceptions.DataConnectorError):
+        # noinspection PyUnusedLocal
+        batch_list: List[
+            Batch
+        ] = basic_datasource_with_runtime_data_connector.get_batch_list_from_batch_request(
+            batch_request=batch_request
+        )
+
+
+def test_set_data_asset_name_for_runtime_data(
+    basic_datasource_with_runtime_data_connector,
+):
+    test_df: pd.DataFrame = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    partition_request: dict
+    partition_request = {
+        "batch_identifiers": {
+            "pipeline_stage_name": "core_processing",
+            "airflow_run_id": 1234567890,
+            "custom_key_0": "custom_value_0",
+        }
+    }
+
+    # set : my_runtime_data_asset
+    batch_request: dict = {
+        "datasource_name": basic_datasource_with_runtime_data_connector.name,
+        "data_connector_name": "test_runtime_data_connector",
+        "data_asset_name": "my_runtime_data_asset",
+        "batch_data": test_df,
+        "partition_request": partition_request,
+        "limit": None,
+    }
+    batch_request: BatchRequest = BatchRequest(**batch_request)
+    batch_list: List[
+        Batch
+    ] = basic_datasource_with_runtime_data_connector.get_batch_list_from_batch_request(
+        batch_request=batch_request
+    )
+    assert batch_list[0].batch_definition.data_asset_name == "my_runtime_data_asset"
+
+
+def test_get_available_data_asset_names(basic_datasource_with_runtime_data_connector):
+    expected_available_data_asset_names: Dict[List[str]] = {
+        "test_runtime_data_connector": []
+    }
+    available_data_asset_names: Dict[
+        List[str]
+    ] = basic_datasource_with_runtime_data_connector.get_available_data_asset_names()
+    assert available_data_asset_names == expected_available_data_asset_names
```

Comment:
@Shinnnyshinshin : Should we not be able to now to have a non-empty list of data_asset_names in the `RuntimeDataConnnector`?..

Reply:
In cases where we have a configuration for `RuntimeDataConnector` and run it through `self_check()` before passing in a `BatchRequest`, the available data_asset_names can be empty.